libyul/AsmAnalysis.cpp

```diff
@@ -101,24 +100,32 @@ AsmAnalysisInfo AsmAnalyzer::analyzeStrictAssertCorrect(Dialect const& _dialect,
 std::vector<YulString> AsmAnalyzer::operator()(Literal const& _literal)
 {
 	expectValidType(_literal.type, nativeLocationOf(_literal));
-	if (_literal.kind == LiteralKind::String && _literal.value.str().size() > 32)
+	bool erroneousLiteralValue = false;
+	if (_literal.kind == LiteralKind::String && !_literal.value.unlimited() && _literal.value.hint() && _literal.value.hint()->size() > 32)
+	{
+		erroneousLiteralValue = true;
 		m_errorReporter.typeError(
 			3069_error,
 			nativeLocationOf(_literal),
-			"String literal too long (" + std::to_string(_literal.value.str().size()) + " > 32)"
+			"String literal too long (" + std::to_string(formatLiteral(_literal, true).size()) + " > 32)"
```

Comment:
Wait, won't the assertion in `formatLiteral()` fail when the string is too long? How it passing tests?
```suggestion
			"String literal too long (" + std::to_string(formatLiteral(_literal, false).size()) + " > 32)"
```

Reply:
this is b/c it is now `assumeValidity` :)

if I assume the validity is given and I trust the flag, then asserts can minimal / be skipped

Reply:
Haha, I didn't expect that :) This is the exact opposite of how I was thinking about it. You assume it is valid, so you can safely assert that. Otherwise it may or may not be valid and the function is fully prepared to handle the invalid cases. Assertion just documents the assumption and does not count as a proper validation (which is part of the reason I didn't really like the idea of it being in the parameter name).

But if this wasn't clear then it's not a good name after all and we need to change it again. I still don't like `_assertValidity`, but I guess it would still be better than something that is ambiguous. My proposal would be to call it `_validated` though.

At this point I'm fine with whichever you choose, not to hold this PR any longer :)